### PR TITLE
Movie poster view with favorite button added

### PR DESCRIPTION
**Pull request by lukan98**

### CODE REVIEW

File: MovieApp/PresentationLayer/Components/MoviePoster/MoviePoster+Design.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    func defineLayoutForViews() {
+        posterImage.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+
+        favoriteButton.snp.makeConstraints {
+            $0.top.leading.equalToSuperview().inset(10)
+            $0.width.height.equalTo(buttonSize)
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Components/MoviePoster/MoviePoster+Design.swift]: Define `let buttonSize = CGSize(width: 32, height: 32)` and then write `$0.size.equalTo(buttonSize)`

File: MovieApp/PresentationLayer/Components/MoviePoster/MoviePoster.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+class MoviePoster: UIView {
+
+    let cornerRadius: CGFloat = 10
+    let buttonSize: CGFloat = 32
+
+    var favorited: Bool = false {
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Components/MoviePoster/MoviePoster.swift]: isFavorite

File: MovieApp/PresentationLayer/Components/MoviePoster/MoviePoster.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        didSet {
+            if favorited {
+                favoriteButton.setImage(UIImage(named: "Favorites-fill"), for: .normal)
+            } else {
+                favoriteButton.setImage(UIImage(named: "Favorites-outline"), for: .normal)
+            }
+        }
+    }
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Components/MoviePoster/MoviePoster.swift:16]: space below

File: MovieApp/PresentationLayer/Favorites/FavoritesViewController+Design.swift [hunk 4]
```diff
@@ -15,6 +15,9 @@ extension FavoritesViewController: ConstructViewsProtocol {
     func createViews() {
         navigationView = NavBarView()
         view.addSubview(navigationView)
+
+        moviePoster = MoviePoster()
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Favorites/FavoritesViewController+Design.swift]: Why did you add this to this VC? For the screenshot or? Because it will probably be in some collection view so delete this.

**Review by istaudacher — CHANGES_REQUESTED** [review]

File: MovieApp/PresentationLayer/Favorites/FavoritesViewController+Design.swift (hunk 4 above)
Comment (lukan98) [reply MovieApp/PresentationLayer/Favorites/FavoritesViewController+Design.swift]: Just for the screenshot and testing, yes 😃

File: MovieApp/PresentationLayer/Favorites/FavoritesViewController.swift [hunk 5]
```diff
@@ -10,4 +10,8 @@ class FavoritesViewController: UIViewController {
         buildViews()
     }
 
+    override func viewDidLayoutSubviews() {
```
Comment (istaudacher) [inline MovieApp/PresentationLayer/Favorites/FavoritesViewController.swift]: Do you need this?

**Review by istaudacher — CHANGES_REQUESTED** [review]

**Review by istaudacher — APPROVED** [review]
